pkg/apis/core/validation/validation.go

```diff
@@ -3435,10 +3435,12 @@ func ValidateAppArmorPodAnnotations(annotations map[string]string, spec *core.Po
 
 func podSpecHasContainer(spec *core.PodSpec, containerName string) bool {
 	var hasContainer bool
-	podshelper.VisitContainersWithPath(spec, func(c *core.Container, _ *field.Path) {
+	podshelper.VisitContainersWithPath(spec, func(c *core.Container, _ *field.Path) bool {
```

Comment:
podSpecHasContainer doesn't actually need path so it could use `VisitContainers` from `pkg/api/pod/util.go` which already has early return. I don't see an import restriction, but I didn't try it.

Reply:
VisitContainers is used now